diablo/models/course_preference.py

```diff
@@ -55,18 +57,20 @@ def get_section_ids_opted_out(cls, term_id):
 
     @classmethod
     def update_opt_out(cls, term_id, section_id, opt_out):
-        preferences = cls.query.filter_by(term_id=term_id, section_id=section_id).first()
-        if preferences:
-            preferences.has_opted_out = opt_out
-        else:
+        section_ids = CrossListing.get_cross_listed_sections(section_id=section_id, term_id=term_id) + [section_id]
+        criteria = and_(cls.section_id.in_(section_ids), cls.term_id == term_id)
+        for row in cls.query.filter(criteria).all():
+            row.has_opted_out = opt_out
+            section_ids.remove(row.section_id)
+        for section_id in section_ids:
```

Comment:
The change above is for sake of https://jira.ets.berkeley.edu/jira/browse/DIABLO-197